data/result-exporter/swe-cne-exporter/src/main/java/com/farao_community/farao/data/swe_cne_exporter/SweConstraintSeriesCreator.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.swe_cne_exporter;
+
+import com.farao_community.farao.data.cne_exporter_commons.CneHelper;
+import com.farao_community.farao.data.cne_exporter_commons.CneUtil;
+import com.farao_community.farao.data.crac_api.Contingency;
+import com.farao_community.farao.data.swe_cne_exporter.xsd.*;
+
+import java.util.*;
+
+import static com.farao_community.farao.data.cne_exporter_commons.CneConstants.B56_BUSINESS_TYPE;
+import static com.farao_community.farao.data.cne_exporter_commons.CneConstants.B57_BUSINESS_TYPE;
+
+/**
+ * Structures the chaining of RASeriesCreator and MonitoredSeriesCreator for SWE CNE format
+ *
+ * @author Philippe Edwards {@literal <philippe.edwards at rte-france.com>}
+ */
+public final class SweConstraintSeriesCreator {
+    private final CneHelper cneHelper;
+    private final SweMonitoredSeriesCreator monitoredSeriesCreator;
+    private final SweRemedialActionSeriesCreator remedialActionSeriesCreator;
+
+    public SweConstraintSeriesCreator(CneHelper cneHelper) {
+        this.cneHelper = cneHelper;
+        this.monitoredSeriesCreator = new SweMonitoredSeriesCreator(cneHelper);
+        this.remedialActionSeriesCreator = new SweRemedialActionSeriesCreator(cneHelper);
+    }
+
+    public List<ConstraintSeries> generate() {
+        List<ConstraintSeries> allConstraintSeries = new ArrayList<>();
+        allConstraintSeries.addAll(generateB56());
+        allConstraintSeries.addAll(generateB57());
+        return allConstraintSeries;
+    }
+
+    private List<ConstraintSeries> generateB56() {
+        List<ConstraintSeries> constraintSeries = new ArrayList<>();
+        constraintSeries.add(generateBasecaseB56());
+        for (Contingency contingency : cneHelper.getCrac().getContingencies()) {
+            constraintSeries.add(generateContingencyB56(contingency));
+        }
```

Comment:
shouldn't you sort the CRAC's contingencies before using them in order to ensure reproducibility ?

Reply:
indeed